integrations/openai/tests/test_toolkit.py

```diff
@@ -0,0 +1,462 @@
+import json
+import logging
+import os
+import uuid
+from contextlib import contextmanager
+from typing import Dict, List, Optional
+from unittest import mock
+
+import openai
+import pytest
+from databricks.sdk.service.catalog import (
+    FunctionInfo,
+    FunctionParameterInfo,
+    FunctionParameterInfos,
+)
+from openai.types.chat.chat_completion_message_tool_call import Function
+from unitycatalog.ai.client import set_uc_function_client
+from unitycatalog.ai.databricks import DatabricksFunctionClient
+from unitycatalog.ai.utils.function_processing_utils import get_tool_name
+
+from tests.helper_functions import mock_chat_completion_response, mock_choice, requires_databricks
+from unitycatalog_ai_openai.toolkit import OpenAIToolkit
+
+CATALOG = "ml"
+SCHEMA = "serena_uc_test"
+USE_SERVERLESS = "USE_SERVERLESS"
+
+_logger = logging.getLogger(__name__)
+
+
+def get_client() -> DatabricksFunctionClient:
+    with mock.patch(
+        "unitycatalog.ai.databricks.get_default_databricks_workspace_client",
+        return_value=mock.Mock(),
+    ):
+        if os.environ.get(USE_SERVERLESS, "false").lower() == "true":
+            return DatabricksFunctionClient()
+        else:
+            return DatabricksFunctionClient(warehouse_id="warehouse_id", cluster_id="cluster_id")
+
+
+def random_func_name():
+    return f"{CATALOG}.{SCHEMA}.test_{uuid.uuid4().hex[:4]}"
+
+
+@contextmanager
+def create_function_and_cleanup(
```

Comment:
looks like each integration has this function. Can we create common utils and import this function from there?

Reply:
The same comments to other helper functions.

Reply:
Then we need to pack them when packing unitycatalog-ai package?

Reply:
what do you mean by `packing`?

Reply:
I mean `building the unitycatalog-ai package`.
If we define this at a place that all sub-pacakges have access to, the only common dependency is `unitycatalog-ai` package.

Reply:
can create a package (e.g. test_utils) in the repository root?

Reply:
You mean under tests/test_utils? Let me try this in a follow-up PR :)